subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/Dependencies.java

```diff
@@ -108,6 +123,59 @@ default ExternalModuleDependency module(@Nullable String group, String name, @Nu
         return getDependencyFactory().create(group, name, version);
     }
 
+    /**
+     * Create a {@link DependencyConstraint} from the given notation.
+     *
+     * @param dependencyConstraintNotation dependency constraint to add
+     * @return the new dependency constraint
+     * @see DependencyConstraintFactory#create(CharSequence) Valid dependency constraint notation for this method
+     */
+    default DependencyConstraint constraint(CharSequence dependencyConstraintNotation) {
+        return getDependencyConstraintFactory().create(dependencyConstraintNotation);
+    }
+
+    /**
+     * Create a {@link DependencyConstraint} from a series of strings.
+     *
+     * @param group the group (optional)
+     * @param name the name
+     * @param version the version (optional)
+     * @return the new dependency constraint
+     */
+    default DependencyConstraint constraint(@Nullable String group, String name, @Nullable String version) {
+        return getDependencyConstraintFactory().create(group, name, version);
+    }
+
+    /**
+     * Create a {@link DependencyConstraint} from a minimal dependency.
+     *
+     * @param dependencyConstraint dependency constraint to add
+     * @return the new dependency constraint
+     */
+    default Provider<? extends DependencyConstraint> constraint(Provider<? extends MinimalExternalModuleDependency> dependencyConstraint) {
+        return dependencyConstraint.map(getDependencyConstraintFactory()::create);
+    }
+
+    /**
+     * Create a {@link DependencyConstraint} from a minimal dependency.
+     *
+     * @param dependencyConstraint dependency constraint to add
+     * @return the new dependency constraint
+     */
+    default Provider<? extends DependencyConstraint> constraint(ProviderConvertible<? extends MinimalExternalModuleDependency> dependencyConstraint) {
+        return constraint(dependencyConstraint.asProvider());
+    }
+
+    /**
+     * Create a {@link DependencyConstraint} from a project.
+     *
+     * @param project the project
+     * @return the new dependency constraint
+     */
+    default DependencyConstraint constraint(ProjectDependency project) {
+        return getDependencyConstraintFactory().create(project.getDependencyProject());
```

Comment:
_ideally_ this would accept some project ID like its identity path/build tree path or a project component identifier. 

But, project dependencies are annoying and actually need a reference to the project in question. I don't see a _huge_ difference here, since a project dependency is really just a wrapper around a project. 

Not sure how much that will change in the future. It would be nice if we didn't need a whole project instance to create a project dependency constraint.

Reply:
Yeah, I can stop unwrapping it. It's already just wrapping it again further down so this doesn't make sense to do.